android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.java

```diff
@@ -302,6 +312,23 @@ private TextView getTitleTextView() {
     return null;
   }
 
+  @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+  protected void applyWindowInsets(Window window) {
+    window.getDecorView().setOnApplyWindowInsetsListener(new View.OnApplyWindowInsetsListener() {
+      @Override
+      public WindowInsets onApplyWindowInsets(View v, WindowInsets insets) {
+        WindowInsets defaultInsets = v.onApplyWindowInsets(insets);
+        // this inset is set to 0 by translucent status bar
+        if (defaultInsets.getStableInsetTop() == 0) {
+          mToolbar.setPadding(0, insets.getSystemWindowInsetTop(), 0, 0);
+        } else {
+          mToolbar.setPadding(0, defaultInsets.getSystemWindowInsetTop(), 0, 0);
+        }
+        return insets.consumeSystemWindowInsets();
```

Comment:
You really consumed only top inset in here, haven't you? There should be a way of marking it.